From: Petr Ovtchenkov <ptr@void-ptr.info>
Date: Thu, 25 Jun 2009 08:06:11 +0000 (+0400)
Subject: Fix build for ARM GNU/Linux
X-Git-Url: http://stlport.git.sourceforge.net/git/gitweb.cgi?p=stlport;a=commitdiff_plain;h=7820fe8e502b062f82d8bb630ad07aa20f6d2d4c

Fix build for ARM GNU/Linux

llabs, lldiv_t, lldiv not defined in native gcc's std namespace;
use ones from C space (i.e. from global namespace) instead.

Build done for Sourcery G++ Lite 2009q1-176 for ARM GNU/Linux;
unit test build success, run not checked.
---

Index: stlport5.2-5.2.1/stlport/stl/_cstdlib.h
===================================================================
--- stlport5.2-5.2.1.orig/stlport/stl/_cstdlib.h	2008-12-10 10:56:51.000000000 +0100
+++ stlport5.2-5.2.1/stlport/stl/_cstdlib.h	2009-06-28 15:22:50.000000000 +0200
@@ -91,7 +91,7 @@
 using _STLP_VENDOR_CSTD::labs;
 using _STLP_VENDOR_CSTD::ldiv;
 #    if defined (_STLP_LONG_LONG) && !defined (_STLP_NO_VENDOR_STDLIB_L)
-#      if !defined(__sun)
+#      if !defined(__sun) && !defined(__arm__)
 using _STLP_VENDOR_CSTD::llabs;
 using _STLP_VENDOR_CSTD::lldiv_t;
 using _STLP_VENDOR_CSTD::lldiv;
@@ -147,7 +147,7 @@
 
 #if defined (_STLP_LONG_LONG)
 #  if !defined (_STLP_NO_VENDOR_STDLIB_L)
-#    if !defined (__sun)
+#    if !defined (__sun) && !defined(__arm__) && !defined(__mips__) && !defined(__mipsel__)
 inline _STLP_LONG_LONG  abs(_STLP_LONG_LONG __x) { return _STLP_VENDOR_CSTD::llabs(__x); }
 inline lldiv_t div(_STLP_LONG_LONG __x, _STLP_LONG_LONG __y) { return _STLP_VENDOR_CSTD::lldiv(__x, __y); }
 #    else
